Add OpenLM LLM multi-provider

r2d4 opened the pull request:
OpenLM is a zero-dependency OpenAI-compatible LLM provider that can call different inference endpoints directly via HTTP. It implements the OpenAI Completion class so that it can be used as a drop-in replacement for the OpenAI API. This changeset utilizes BaseOpenAI for minimal added code.

dev2049 commented:
thanks @r2d4! could we add a simple example notebook (a la docs/modules/models/llms/integrations/openai.ipynb)

dev2049 commented on pyproject.toml:
```diff
@@ -90,7 +90,7 @@ pandas = {version = "^2.0.1", optional = true}
 zep-python = {version="^0.25", optional=true}
 chardet = {version="^5.1.0", optional=true}
-
+openlm = {version = "^0.0.5", optional = true}
```
if this isn't added to one of the extras it'll be installed automatically. could we either delete or add to "all" (line 186)

r2d4 committed:
Add OpenLM LLM multi-provider

OpenLM is a zero-dependency OpenAI-compatible LLM provider that can call different inference endpoints directly via HTTP. It implements the OpenAI Completion class so that it can be used as a drop-in replacement for the OpenAI API. This changeset utilizes BaseOpenAI for minimal added code.

r2d4 commented:
Thanks for the feedback @dev2049

- Added documentation in the form of a notebook
- Added the dependency to extras so it isn't downloaded automatically

Let me know if there are any other changes!

r2d4 committed:
resolve lockfile

r2d4 committed:
format

dev2049 committed:
fmt

dev2049 committed:
cr

dev2049 committed:
poetry

dev2049 committed:
clean
